### DISCUSSION TITLE
SETTINGS_MODULE not set in tests

### DISCUSSION THREAD

**Discussion by CleitonDeLima** [django-components/django-components#332] [Q&A] [answered]
I came across this [case](https://github.com/EmilStenstrom/django-components/commit/f5c3f64c866d06dfa2435d1d18b936dddb9eb67c) in my tests, when modifying a `settings` using the `pytest-django` fixture, my `SETTINGS_MODULE` becomes `None` and raises `TemplateDoesNotExist`. 
Did this happen to anyone?

**Comment by EmilStenstrom** [comment]
That's an interesting case. Where would you expect django-components to look for your components if the settings file does not exist?

**Reply by CleitonDeLima** [reply]
My test is more or less like this:

```python
def test_home_view(client, settings):
        settings.FOO = "bar"
		url = resolve_url("home")
       
        response = client.get(url)
        assert response.status_code == 200
```

The home view renders a component, but the error occurs:

```sh
def reraise(exc, backend):
        """
        Reraise TemplateDoesNotExist while maintaining template debug information.
        """
        new = copy_exception(exc, backend)
>       raise new from exc
E       django.template.exceptions.TemplateDoesNotExist: table/table.html
```

It seems to me that it's a problem with `pytest-django` when I try to change something in the `settings`. Don't look too deeply into the problem.

> That's an interesting case. Where would you expect django-components to look for your components if the settings file does not exist?

The issue is that the `SETTINGS_MODULE` variable does not exist during this test, other settings information I can access normally. 

If the path to the components folder was configurable instead of `SETTINGS_MODULE`, would this be viable?

```python
COMPONENTS = {
    "components_path": BASE_DIR / "components",
}
```

**Reply by EmilStenstrom** [reply]
It IS configurable! https://github.com/EmilStenstrom/django-components#configure-the-module-where-components-are-loaded-from

... not sure how it handles SETTINGS_MODULE not defined though. Mind testing and reporting back?

**Comment by CleitonDeLima** [comment]
> It IS configurable! [EmilStenstrom/django-components#configure-the-module-where-components-are-loaded-from](https://github.com/EmilStenstrom/django-components#configure-the-module-where-components-are-loaded-from)
> 
> ... not sure how it handles SETTINGS_MODULE not defined though. Mind testing and reporting back?

Using this configuration I get the same problem...

`Loader.get_dirs` don't read `COMPONENTS.libraries` in settings?

**Comment by EmilStenstrom** [comment, marked as answer]
This is now solved in 0.35 of django-components, by reading the BASE_DIRS setting. Thanks @dylanjcastillo!
